In [3]:
import pandas as pd

In [4]:
df = pd.read_csv("pokemon.csv")

We learned to how to import a file in importing, Now we learn how to select a particular set of elements from a column. (SELECTION VIA COLUMN)

In [5]:
print(df["Name"])

0                  Bulbasaur
1                    Ivysaur
2                   Venusaur
3      VenusaurMega Venusaur
4                 Charmander
               ...          
795                  Diancie
796      DiancieMega Diancie
797      HoopaHoopa Confined
798       HoopaHoopa Unbound
799                Volcanion
Name: Name, Length: 800, dtype: str


The default print for a very large set of data is: Pandas chooses 5 element of the top and the 5 element at the last.

In [6]:
#The above displayed table is a trunketed version. 
#To display upto a selected number we use:
print(df["Name"].head(25))

0                     Bulbasaur
1                       Ivysaur
2                      Venusaur
3         VenusaurMega Venusaur
4                    Charmander
5                    Charmeleon
6                     Charizard
7     CharizardMega Charizard X
8     CharizardMega Charizard Y
9                      Squirtle
10                    Wartortle
11                    Blastoise
12      BlastoiseMega Blastoise
13                     Caterpie
14                      Metapod
15                   Butterfree
16                       Weedle
17                       Kakuna
18                     Beedrill
19        BeedrillMega Beedrill
20                       Pidgey
21                    Pidgeotto
22                      Pidgeot
23          PidgeotMega Pidgeot
24                      Rattata
Name: Name, dtype: str


In [7]:
#                       To select multiple columnes:
print(df[["Name", "Attack", "Defense"]].head(26))  # df["height"].  the [] is called the subscript operator.

#Note:
#   To print multiple columnes we send a list of columes we want to see
#   in the subscript operator, the subscript operator expects a single
#   string or a list of items.

                         Name  Attack  Defense
0                   Bulbasaur      49       49
1                     Ivysaur      62       63
2                    Venusaur      82       83
3       VenusaurMega Venusaur     100      123
4                  Charmander      52       43
5                  Charmeleon      64       58
6                   Charizard      84       78
7   CharizardMega Charizard X     130      111
8   CharizardMega Charizard Y     104       78
9                    Squirtle      48       65
10                  Wartortle      63       80
11                  Blastoise      83      100
12    BlastoiseMega Blastoise     103      120
13                   Caterpie      30       35
14                    Metapod      20       55
15                 Butterfree      45       50
16                     Weedle      35       30
17                     Kakuna      25       50
18                   Beedrill      90       40
19      BeedrillMega Beedrill     150       40
20           

SELECTION BY ROW/S

In [8]:
print(df.loc[0])  #The 0 is a by default assignment by pandas to act as a label.
#                   similar to array index it starts with 0, and in this case prints info about bulbasaur.

#                     1
Name          Bulbasaur
Type 1            Grass
Type 2           Poison
Total               318
HP                   45
Attack               49
Defense              49
Sp. Atk              65
Sp. Def              65
Speed                45
Generation            1
Legendary         False
Name: 0, dtype: object


Now here arises a problem:

You aren't going to remember all 800 pokemon by index from 0 to 799. So we assign a columne to act as a index and in this case we want the name of pokemon to act as index, So we can access all the data regarding a particular pokemon via its name.

In [9]:
pokedex = pd.read_csv("pokemon.csv", index_col="Name")
print(df.head(26))

     #                       Name  Type 1  Type 2  Total  HP  Attack  Defense  \
0    1                  Bulbasaur   Grass  Poison    318  45      49       49   
1    2                    Ivysaur   Grass  Poison    405  60      62       63   
2    3                   Venusaur   Grass  Poison    525  80      82       83   
3    3      VenusaurMega Venusaur   Grass  Poison    625  80     100      123   
4    4                 Charmander    Fire     NaN    309  39      52       43   
5    5                 Charmeleon    Fire     NaN    405  58      64       58   
6    6                  Charizard    Fire  Flying    534  78      84       78   
7    6  CharizardMega Charizard X    Fire  Dragon    634  78     130      111   
8    6  CharizardMega Charizard Y    Fire  Flying    634  78     104       78   
9    7                   Squirtle   Water     NaN    314  44      48       65   
10   8                  Wartortle   Water     NaN    405  59      63       80   
11   9                  Blas

In [ ]:
print(pokedex.loc["Bulbasaur"])  #.loc means location by index
#                            .iloc or integer location is position based(integer index)

#                  1
Type 1         Grass
Type 2        Poison
Total            318
HP                45
Attack            49
Defense           49
Sp. Atk           65
Sp. Def           65
Speed             45
Generation         1
Legendary      False
Name: Bulbasaur, dtype: object


In [11]:
print(pokedex.loc["Dialga"])

#                483
Type 1         Steel
Type 2        Dragon
Total            680
HP               100
Attack           120
Defense          120
Sp. Atk          150
Sp. Def          100
Speed             90
Generation         4
Legendary       True
Name: Dialga, dtype: object


In [12]:
print(pokedex.loc["Charizard"])

#                  6
Type 1          Fire
Type 2        Flying
Total            534
HP                78
Attack            84
Defense           78
Sp. Atk          109
Sp. Def           85
Speed            100
Generation         1
Legendary      False
Name: Charizard, dtype: object


In [14]:
# If we don't want all the columne in the specific row, we can 
#  send in a list of columnes we would like to see, alongside the index in
#   the .loc function.
print(pokedex.loc["Zekrom", ["Total", "Attack", "Speed"]])

Total     680
Attack    150
Speed      90
Name: Zekrom, dtype: int64


In [16]:
#SELECTION OF PARTICULAR RANGE OF ROW:
#                           We do this via using a slice operator ':'
print(pokedex.loc["Bulbasaur":"Charizard", ["Total", "Attack", "Speed"]])

                       Total  Attack  Speed
Name                                       
Bulbasaur                318      49     45
Ivysaur                  405      62     60
Venusaur                 525      82     80
VenusaurMega Venusaur    625     100     80
Charmander               309      52     65
Charmeleon               405      64     80
Charizard                534      84    100


In [19]:
print(pokedex.iloc[0:11])    #slicing to display the first 10 pokemon.

                           # Type 1  Type 2  Total  HP  Attack  Defense  \
Name                                                                      
Bulbasaur                  1  Grass  Poison    318  45      49       49   
Ivysaur                    2  Grass  Poison    405  60      62       63   
Venusaur                   3  Grass  Poison    525  80      82       83   
VenusaurMega Venusaur      3  Grass  Poison    625  80     100      123   
Charmander                 4   Fire     NaN    309  39      52       43   
Charmeleon                 5   Fire     NaN    405  58      64       58   
Charizard                  6   Fire  Flying    534  78      84       78   
CharizardMega Charizard X  6   Fire  Dragon    634  78     130      111   
CharizardMega Charizard Y  6   Fire  Flying    634  78     104       78   
Squirtle                   7  Water     NaN    314  44      48       65   
Wartortle                  8  Water     NaN    405  59      63       80   

                        

In [ ]:
#If we need every ODD no. of rows within a partcular set of rows from 1 to 10
print(pokedex.iloc[0:11:2])

                           # Type 1  Type 2  Total  HP  Attack  Defense  \
Name                                                                      
Bulbasaur                  1  Grass  Poison    318  45      49       49   
Venusaur                   3  Grass  Poison    525  80      82       83   
Charmander                 4   Fire     NaN    309  39      52       43   
Charizard                  6   Fire  Flying    534  78      84       78   
CharizardMega Charizard Y  6   Fire  Flying    634  78     104       78   
Wartortle                  8  Water     NaN    405  59      63       80   

                           Sp. Atk  Sp. Def  Speed  Generation  Legendary  
Name                                                                       
Bulbasaur                       65       65     45           1      False  
Venusaur                       100      100     80           1      False  
Charmander                      60       50     65           1      False  
Charizard          

In [22]:
#Similarly for ODD no. of rows
print(pokedex.iloc[1:11:2])

                           # Type 1  Type 2  Total  HP  Attack  Defense  \
Name                                                                      
Ivysaur                    2  Grass  Poison    405  60      62       63   
VenusaurMega Venusaur      3  Grass  Poison    625  80     100      123   
Charmeleon                 5   Fire     NaN    405  58      64       58   
CharizardMega Charizard X  6   Fire  Dragon    634  78     130      111   
Squirtle                   7  Water     NaN    314  44      48       65   

                           Sp. Atk  Sp. Def  Speed  Generation  Legendary  
Name                                                                       
Ivysaur                         80       80     60           1      False  
VenusaurMega Venusaur          122      120     80           1      False  
Charmeleon                      80       65     80           1      False  
CharizardMega Charizard X      130       85    100           1      False  
Squirtle          

In [23]:
#For second argument in iloc we can select indices of which columns we would like:
print(pokedex.iloc[0:11:2, 0:3])    #I would like to print first 3 columne

                           # Type 1  Type 2
Name                                       
Bulbasaur                  1  Grass  Poison
Venusaur                   3  Grass  Poison
Charmander                 4   Fire     NaN
Charizard                  6   Fire  Flying
CharizardMega Charizard Y  6   Fire  Flying
Wartortle                  8  Water     NaN


Lets create a pokedex that asks the user about a pokemon and prints info regarding it.

In [26]:
pokemon = input("Enter a Pokemon name: ")

try:
    print(pokedex.loc[pokemon])
except KeyError:
    print(f"{pokemon} not found")

cat not found
